xbmc/platform/android/activity/JNIXBMCRecognitionListener.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "JNIXBMCRecognitionListener.h"
+
+#include "CompileInfo.h"
+#include "ServiceBroker.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindow.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "messaging/ApplicationMessenger.h"
+#include "utils/log.h"
+
+#include <androidjni/Context.h>
+#include <androidjni/jutils-details.hpp>
+
+using namespace jni;
+
+static std::string s_className =
+    std::string(CCompileInfo::GetClass()) + "/interfaces/XBMCRecognitionListener";
+
+CJNIXBMCRecognitionListener::CJNIXBMCRecognitionListener() : CJNIBase(s_className)
+{
+  m_object = new_object(CJNIContext::getClassLoader().loadClass(GetDotClassName(s_className)));
+  m_object.setGlobal();
+
+  add_instance(m_object, this);
+}
+
+void CJNIXBMCRecognitionListener::RegisterNatives(JNIEnv* env)
+{
+  jclass cClass = env->FindClass(s_className.c_str());
+  if (cClass)
+  {
+    JNINativeMethod methods[] = {
+        {"_onReadyForSpeech", "(Landroid/os/Bundle;)V",
+         reinterpret_cast<void*>(&CJNIXBMCRecognitionListener::_onReadyForSpeech)},
+        {"_onError", "(I)V", reinterpret_cast<void*>(&CJNIXBMCRecognitionListener::_onError)},
+        {"_onResults", "(Landroid/os/Bundle;)V",
+         reinterpret_cast<void*>(&CJNIXBMCRecognitionListener::_onResults)},
+    };
+
+    env->RegisterNatives(cClass, methods, sizeof(methods) / sizeof(methods[0]));
+  }
+}
+
+CJNIXBMCRecognitionListener::~CJNIXBMCRecognitionListener()
+{
+  remove_instance(this);
+}
+
+void CJNIXBMCRecognitionListener::_onReadyForSpeech(JNIEnv* env, jobject thiz, jobject bundle)
+{
+  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, g_localizeStrings.Get(39177),
+      g_localizeStrings.Get(39179));
```

Comment:
Same approach as for ::_onError. define a new message TMSG_SPEECH_RECOGNIZER_READY, disapach via AppMessenger to Capp, then to `CGUIDialogKeyboardGeneric::OnSpeechRecognizerReady`, which displays the toast.